maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java

```diff
@@ -1143,6 +1204,29 @@ private boolean validateBannedCharacters( String fieldName, ModelProblemCollecto
         return true;
     }
 
+    @SuppressWarnings( "checkstyle:parameternumber" )
+    private boolean validateBannedCharacters( String prefix, String fieldName, ModelProblemCollector problems,
+                                              Severity severity, Version version, String string, String sourceHint,
+                                              InputLocationTracker tracker, String banned )
+    {
+        if ( string != null )
+        {
+            for ( int i = string.length() - 1; i >= 0; i-- )
```

Comment:
Why do you reverse iterate the string?

Reply:
This isn't new code, I'll push a new commit that modifies the existing method instead of duplicating it.